olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -379,24 +400,62 @@ uint64_t DefaultCacheImpl::MaybeEvictData(leveldb::WriteBatch& batch) {
   int64_t evicted = 0u;
   auto count = 0u;
   const auto min_size = kMinDiskUsedThreshold * settings_.max_disk_storage;
+
+  const auto current_time = olp::cache::InMemoryCache::DefaultTimeProvider()();
+
+  // Remove the expired elements first
+  for (auto it = mutable_cache_lru_->begin();
```

Comment:
Maybe start from rbegin(), same as below and search for expired. After a certain point i guess the elements that are at the end are more likely to be expired.